### PR TITLE
Miscellaneous changes

### PR DESCRIPTION
**Pull request by wkaluza**

### CODE REVIEW

File: libs/core/include/core/serializers/binary_interface.hpp [hunk 1]
```diff
@@ -76,34 +82,34 @@ class BinaryConstructorInterface
...
 
-      uint8_t size = static_cast<uint8_t>(count);
-      size         = platform::ToBigEndian(size);
+      auto size = static_cast<uint8_t>(count);
+      size      = platform::ToBigEndian(size);
       serializer_.Allocate(sizeof(size));
       serializer_.WriteBytes(&size, sizeof(size));
     }
-    else if (count < (1 << 16))
+    else if (count < (1u << 16u))
```
Comment (bipll) [inline libs/core/include/core/serializers/binary_interface.hpp]: The three boundaries in this `if` statement are, in succession, `1 << 8`, `1u << 16u` and `1ull << 32`. Pick a single format perhaps?

File: libs/core/include/core/serializers/binary_interface.hpp [hunk 2]
```diff
@@ -76,34 +82,34 @@ class BinaryConstructorInterface
...
     {
-      uint8_t opcode = static_cast<uint8_t>(CODE16);
+      auto opcode = static_cast<uint8_t>(CODE16);
       serializer_.Allocate(sizeof(opcode));
       serializer_.WriteBytes(&opcode, sizeof(opcode));
 
-      uint16_t size = static_cast<uint16_t>(count);
-      size          = platform::ToBigEndian(size);
+      auto size = static_cast<uint16_t>(count);
+      size      = platform::ToBigEndian(size);
```
Comment (bipll) [inline libs/core/include/core/serializers/binary_interface.hpp:101]: Can this call be made in the previous line? Still clearly readable yet not introducing (even trivial and ephemeral) recursive dependencies.

File: libs/core/include/core/state_machine.hpp [hunk 3]
```diff
@@ -47,13 +47,14 @@ namespace core {
 template <typename State>
 class StateMachine : public StateMachineInterface, public Runnable
 {
-public:
+private:
```
Comment (bipll) [inline libs/core/include/core/state_machine.hpp]: Is explicit `private` required here?

**Review by bipll — CHANGES_REQUESTED** [review]

File: libs/core/include/core/serializers/binary_interface.hpp (hunk 1 above)
Comment (wkaluza) [reply libs/core/include/core/serializers/binary_interface.hpp]: done

File: libs/core/include/core/serializers/binary_interface.hpp (hunk 2 above)
Comment (wkaluza) [reply libs/core/include/core/serializers/binary_interface.hpp:101]: I think it's more readable the way it is. Room for personal taste.

There is no recursion here.

File: libs/core/include/core/state_machine.hpp (hunk 3 above)
Comment (wkaluza) [reply libs/core/include/core/state_machine.hpp]: Fair, it's not.

**Comment by ejfitzgerald** [comment]
Reviewed after the merge

**Pull request merged**